pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -2642,6 +2649,27 @@ func maybeResizeDisk(diskfile string, maxsizebytes uint64) error {
 	return err
 }
 
+// Fetch the list of environment variables from the cloud init
+// We are expecting the environment variables to be pass in particular format in cloud-int
+// Example:
+// Key1:Val1
+// Key2:Val2 ...
+func fetchEnvVariablesFromCloudInit(config types.DomainConfig) (map[string]string, error) {
+	ud, err := base64.StdEncoding.DecodeString(*config.CloudInitUserData)
+	if err != nil {
+		errStr := fmt.Sprintf("fetchEnvVariablesFromCloudInit failed %s\n", err)
+		return nil, errors.New(errStr)
+	}
+	envList := make(map[string]string, 0)
+	list := strings.Split(string(ud), "\n")
+	for _, v := range list {
+		pair := strings.Split(v, ":")
+		envList[pair[0]] = pair[1]
+	}
```

Comment:
Are you assuming the whole data from cloud-init is only env variables? Hmm.. Ok for now, but we may have to revisit. Lets keep it for now.